File: backend/app/schemas/stock.py
from pydantic import BaseModel
from typing import Optional


class StockKlineRequest(BaseModel):
    period: str = "daily"  # daily, weekly, monthly
    start_date: str
    end_date: str
    adjust: str = "qfq"  # qfq=前复权, hfq=后复权, none=不复权


class StockRealtimeResponse(BaseModel):
    code: str
    name: str
    price: float
    change: float
    change_pct: float
    open: float
    high: float
    low: float
    pre_close: float
    volume: int
    amount: float
    update_time: str


class StockInfoResponse(BaseModel):
    code: str
    name: str
    market: str
    pe: Optional[float] = None
    pb: Optional[float] = None
    market_cap: Optional[float] = None
    total_shares: Optional[int] = None


class IndexDataResponse(BaseModel):
    name: str
    code: str
    price: float
    change: float
    change_pct: float

File: frontend/src/types/stock.ts
export interface KlineData {
  date: string;
  open: number;
  close: number;
  high: number;
  low: number;
  volume: number;
  amount: number;
  amplitude: number;
  change_pct: number;
  change_amount: number;
  turnover: number;
}

export interface RealtimeQuote {
  code: string;
  name: string;
  price: number;
  change: number;
  change_pct: number;
  open: number;
  high: number;
  low: number;
  pre_close: number;
  volume: number;
  amount: number;
  update_time: string;
}

export interface StockInfo {
  code: string;
  name: string;
  market: string;
  pe?: number;
  pb?: number;
  market_cap?: number;
  total_shares?: number;
}

export interface IndexData {
  name: string;
  code: string;
  price: number;
  change: number;
  change_pct: number;
}

export interface StockLinks {
  eastmoney: string;
  tonghuashun_web: string;
  xueqiu: string;
  eastmoney_app: string;
  tonghuashun_app: string;
  xueqiu_app: string;
}
